tests/unit/tsx.ts

```diff
@@ -0,0 +1,63 @@
+import * as registerSuite from 'intern!object';
+import * as assert from 'intern/chai!assert';
+import { WidgetBase } from './../../src/WidgetBase';
+import { HNode, WidgetProperties, WNode } from '../../src/interfaces';
+import { tsx, fromRegistry, REGISTRY_ITEM } from '../../src/tsx';
+import { HNODE } from './../../src/d';
+
+registerSuite({
+	name: 'tsx',
+	'create a registry wrapper'() {
+		const RegistryWrapper = fromRegistry<WidgetProperties>('tag');
+		assert.strictEqual((<any> RegistryWrapper).type, REGISTRY_ITEM);
+		const registryWrapper = new RegistryWrapper();
+		assert.strictEqual(registryWrapper.name, 'tag');
+		// These will always be undefined but show the type inference.
```

Comment:
"show the type inference" ... I'd be more explicit, e.g. "show the type inference of ___________"